src/src_main/crypto_tc.c

```diff
@@ -462,12 +462,14 @@ int32_t Crypto_TC_ApplySecurity(const uint8_t* p_in_frame, const uint16_t in_fra
 
             if(ecs_is_aead_algorithm == CRYPTO_TRUE)
             {
+                crypto_key_t* ek_ring = cryptography_if->get_ek_ring();
+
                 status = cryptography_if->cryptography_aead_encrypt(&p_new_enc_frame[index],                               // ciphertext output
                                                                     (size_t)tf_payload_len,                                        // length of data
                                                                     (uint8_t*)(p_in_frame + TC_FRAME_HEADER_SIZE + segment_hdr_len), // plaintext input
                                                                     (size_t)tf_payload_len,                                         // in data length
                                                                     NULL, // Using SA key reference, key is null
-                                                                    KEY_SIZE, // Length of key. TODO - why is this hard-coded?
+                                                                    ek_ring[sa_ptr->ekid].key_len, // Length of key derived from sa_ptr key_ref
```

Comment:
I thought we agreed in our weekly meetings that key length would be a function of the ECS/ACS algorithm enums being chosen -- EG, if ecs is CRYPTO_CIPHER_AES256_GCM, then that means we have a 32 byte key. (256/8 = 32). AES256 must have a 32 byte key, since we know that's the algorithm, we don't need to reference the ek_ring.

Reply:
<not duplicating my comments for all new key_len functions in this PR, this applies anywhere get_ek_ring() is used here>